blockchain_connector/avalon_blockchain_connector/fabric/fabric_connector.py

```diff
@@ -218,3 +163,110 @@ def workorder_event_handler_func(self, event, block_num, txn_id, status):
                 logging.info("work_order_get_result is failed")
         else:
             logging.info("work_order_submit is failed")
+
+    def _lookup_workers_in_kv_storage(self):
+        """
+        Retrieves the worker ids from shared kv using
+        worker_lookup direct API.
+        Returns list of worker id
+        """
+        jrpc_req_id = random.randint(0, 100000)
+
+        worker_lookup_result = self.__jrpc_worker.worker_lookup(
+            worker_type=WorkerType.TEE_SGX, id=jrpc_req_id
+        )
+        logging.info("\nWorker lookup response from kv storage : {}\n".format(
+            json.dumps(worker_lookup_result, indent=4)
+        ))
+        if "result" in worker_lookup_result and \
+                "ids" in worker_lookup_result["result"].keys():
+            if worker_lookup_result["result"]["totalCount"] != 0:
+                return worker_lookup_result["result"]["ids"]
+            else:
+                logging.error("No workers found in kv storage")
+        else:
+            logging.error("Failed to lookup worker in kv storage")
+        return []
+
+    def _retrieve_worker_details_from_kv_storage(self, worker_id):
+        """
+        Retrieve worker details from shared kv using
+        direct json rpc API
+        Returns the worker details in json string format
+        """
+        jrpc_req_id = random.randint(0, 100000)
+        worker_info = self.__jrpc_worker.worker_retrieve(
+            worker_id, jrpc_req_id)
+        logging.info("Worker retrieve response from kv storage: {}"
+                     .format(json.dumps(worker_info, indent=4)))
+
+        if "error" in worker_info:
+            logging.error("Unable to retrieve worker details from kv storage")
+            return ""
+        else:
+            return worker_info["result"]
+
+    def _lookup_workers_onchain(self):
+        """
+        Lookup all workers on chain to sync up with kv storage
+        Return list of worker ids
+        """
+        worker_lookup_result = self.__fabric_worker.worker_lookup(
```

Comment:
In a later PR we should try to create an common abstract class which can take blockchain and direct instance as argument to the function as the logic remains same for fabric/ethereum blockchain